src/error/go.rs

```diff
@@ -65,3 +66,27 @@ impl fmt::Display for GoResult {
         }
     }
 }
+
+impl GoResult {
+    /// this is like try_from but uses a fallback for GoResult::Other.
+    /// if output is not-empty, use that as the error message
+    /// otherwise, call default() to generate a default message
+    ///
+    /// Safety: this reads data from an externally provided buffer and assumes valid utf-8 encoding
+    /// Only call if you trust the code that provides output to be correct
+    pub unsafe fn into_ffi_result<F>(self, output: Buffer, default: F) -> Result<(), FfiError>
+    where
+        F: Fn() -> String,
+    {
+        self.try_into().unwrap_or_else(|_| {
+            let msg = if output.ptr.is_null() {
+                default()
+            } else {
+                // We initialize `output` with a null pointer. if it is not null,
+                // that means it was initialized by the go code, with values generated by `memory::allocate_rust`
+                String::from_utf8_unchecked(output.consume())
```

Comment:
Can we use `String::from_utf8_lossy` instead? This is safe and should do the same in almost all cases.

Reply:
Sure. I do prefer to drop characters rather than allow bad chars in.
I could also do `String::from_utf(output.consume()).unwrap()` and just panic when go sends bad data (which should never happen, so probably good to fail early)

This will not make the function safe however, as `output.consume()` is already unsafe.

Reply:
String::from_utf8_lossy uses replacement characters, which I think is good here

> During this conversion, from_utf8_lossy() will replace any invalid UTF-8 sequences with U+FFFD REPLACEMENT CHARACTER, which looks like this: �

Panic would also be fine.